www/js/lib/zimArchiveLoader.js

```diff
@@ -70,12 +71,12 @@ define(['zimArchive', 'jquery', 'uiUtil'],
         });
         jQuery.when.apply(null, promises).then(function() {
             callbackFunction(directories);
-        }, function(error) {
-            uiUtil.systemAlert("Error scanning your device storage : " + error
+        }).catch(function (error) {
+            callbackError("Error scanning your device storage : " + error
             + ". If you're using the Firefox OS Simulator, please put the archives in "
             + "a 'fake-sdcard' directory inside your Firefox profile "
             + "(ex : ~/.mozilla/firefox/xxxx.default/extensions/fxos_2_x_simulator@mozilla.org/"
-            + "profile/fake-sdcard/wikipedia_en_ray_charles_2015-06.zim)", "Error reading Device Storage").then(function() { callbackFunction(null); });
+            + "profile/fake-sdcard/wikipedia_en_ray_charles_2015-06.zim)", "Error reading Device Storage").then(function () {callbackFunction(null);});
```

Comment:
Here, will chaining the `.then` block to the `callbackError` work? I am skeptical here as the `callbackError` is itself a function that is calling `uiUtil.systemAlert` which returns a promise. Should I do a `return uiUtil.systemAlert()` inside the `callbackError` I am passing in `app.js`

Reply:
You're right: this will certainly not work.
`callbackError` is a regular function, not a Promise. So there is probably no `.then` function on it.

Previously, we had no error callback. When there was an error, we were calling the normal callback with null value.
Now that we have an error callback, I think we should simply call it, and remove the `.then` here.

However, I'm wondering if we should not add something else in its caller (line 977 of app.js).
To be safe, after calling `uiUtil.systemAlert` (in its `.then`), we should call `populateDropDownListOfArchives(null)`, to have the exact same behavior as before.

Reply:
This is also a FirefoxOS-specific use-case. So I don't expect you to be able to test it.
I'll test myself when you'll request a review for the PR